manifests/service.pp

```diff
@@ -4,21 +4,21 @@
 class candlepin::service {
   assert_private()
 
-  service { 'tomcat':
-    ensure     => running,
-    enable     => true,
-    hasstatus  => true,
-    hasrestart => true,
+  if $candlepin::use_container {
+    file { '/etc/containers/systemd/tomcat.container':
+      ensure  => file,
+      content => template('candlepin/candlepin.container'),
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0444',
+      before  => Service['tomcat'],
+    }
   }
 
-  if $candlepin::run_init {
-    exec { 'cpinit':
-      # tomcat startup is slow - try multiple times (the initialization service is idempotent)
-      command => "/usr/bin/wget --no-check-certificate --no-proxy --timeout=30 --tries=40 --wait=20 --retry-connrefused -qO- https://localhost:${candlepin::ssl_port}/candlepin/admin/init > /var/log/candlepin/cpinit.log 2>&1 && touch /var/lib/candlepin/cpinit_done",
-      require => [Package['wget'], Service['tomcat']],
-      creates => '/var/lib/candlepin/cpinit_done',
-      # timeout is roughly "wait" * "tries" from above
-      timeout => 800,
-    }
+  service { 'tomcat':
+    ensure   => 'running',
+    enable   => true,
+    restart  => true,
+    provider => 'systemd',
```

Comment:
I'd think these are redundant

Reply:
IIRC I needed this to and the change to the definition of the service to prevent some automatic things puppet service type was trying to do.

Reply:
I was explicitly talking about `restart` and `provider`. If it can't detect the proper provider, I'd be worried.

For restart, I'm not sure what `true` does. https://www.puppet.com/docs/puppet/7/type.html#service-attribute-restart isn't clear about that.